### PR TITLE
Adding support for extension to register transport actions

### LINKED ISSUE #119: [FEATURE] Add an interface to help HelloWorld extension to register transport action
### Is your feature request related to a problem?
Now with get Actions support from PR https://github.com/opensearch-project/opensearch-sdk-java/pull/115, we need to expose an interface for HelloWorld extension to register transport actions.

### PR DESCRIPTION
**Pull request by saratvemulapalli**
Signed-off-by: Sarat Vemulapalli <vemulapallisarat@gmail.com>

### Description
Adding support for hello world extension to write up custom transport actions.

### Issues Resolved
#119 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/sdk/ExtensionsRunner.java [hunk 1]
```diff
@@ -126,6 +121,7 @@ public ExtensionsRunner() throws IOException {
             .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort())
             .build();
         this.customSettings = Collections.emptyList();
+        this.transportActions = new TransportActions(Collections.EMPTY_MAP);
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/sdk/ExtensionsRunner.java:124]: Prefer `Collections.emptyMap()` here (which just wraps `EMPTY_MAP` with an inferred `<T>` cast) which is friendlier to code/IDEs and avoids downstream unchecked cast warnings.

**Review by dbwiddis — DISMISSED** [review]
Approved with a trivial suggestion that will probably make no difference in reality.

**Review by owaiskazi19 — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
This PR is waiting on : https://github.com/opensearch-project/OpenSearch/pull/4598

**Review by dbwiddis — APPROVED** [review]
Resolved conflicts. This was previously approved by myself and @owaiskazi19 and needs to get merged ASAP as the build is broken and is blocking #172

**Review by owaiskazi19 — APPROVED** [review]

**Comment by saratvemulapalli** [comment]
> Resolved conflicts. This was previously approved by myself and @owaiskazi19 and needs to get merged ASAP as the build is broken and is blocking #172 

Thanks @dbwiddis for resolving the conflicts.

**Pull request merged**